File: Microsoft Office Developer Documentation Team/SharePoint-2010-Creating-Sequential-Workflows-in-4026/[C#]-SharePoint 2010 Creating Sequential Workflows in Visual Studio 2010/C#/seqWorkflow/Workflow1/Workflow1.cs
using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Linq;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;
using Microsoft.SharePoint;
using Microsoft.SharePoint.Workflow;
using Microsoft.SharePoint.WorkflowActions;

namespace seqWorkflow.Workflow1
{
    public sealed partial class Workflow1 : SequentialWorkflowActivity
    {
        public Workflow1()
        {
            InitializeComponent();
        }

        Boolean bIsWorkflowPending = true;

        public Guid workflowId = default(System.Guid);
        public SPWorkflowActivationProperties workflowProperties = new SPWorkflowActivationProperties();

        private void onWorkflowActivated(object sender, ExternalDataEventArgs e)
        {
            CheckStatus();
        }

        private void isWorkflowPending(object sender, ConditionalEventArgs e)
        {
            e.Result = bIsWorkflowPending;
        }

        private void onWorkflowItemChanged(object sender, ExternalDataEventArgs e)
        {
            CheckStatus();
        }
        private void CheckStatus()
        { 
        if ( workflowProperties.Item["Document Status"].ToString() == "Review Complete")
        {
            bIsWorkflowPending = false;
        }
        }
    }
}

File: Microsoft Office Developer Documentation Team/SharePoint-2010-Creating-Sequential-Workflows-in-4026/[VB]-SharePoint 2010 Creating Sequential Workflows in Visual Studio 2010/VB/seqWorkflow/Workflow1/Workflow1.vb
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Drawing
Imports System.Workflow.ComponentModel.Compiler
Imports System.Workflow.ComponentModel.Serialization
Imports System.Workflow.ComponentModel
Imports System.Workflow.ComponentModel.Design
Imports System.Workflow.Runtime
Imports System.Workflow.Activities
Imports System.Workflow.Activities.Rules
Imports Microsoft.SharePoint.Workflow
Imports Microsoft.SharePoint.WorkflowActions

'NOTE: When changing the namespace; please update XmlnsDefinitionAttribute in AssemblyInfo.vb
Public Class Workflow1
    Inherits SequentialWorkflowActivity
    Public Sub New()
        MyBase.New()
        InitializeComponent()
    End Sub

    Dim bIsWorkflowPending As Boolean = True

    Public workflowProperties As SPWorkflowActivationProperties = New Microsoft.SharePoint.Workflow.SPWorkflowActivationProperties

    Private Sub onWorkflowActivated(ByVal sender As System.Object, ByVal e As System.Workflow.Activities.ExternalDataEventArgs)
        CheckStatus()
    End Sub

    Private Sub isWorkflowPending(ByVal sender As System.Object, ByVal e As System.Workflow.Activities.ConditionalEventArgs)
        e.Result = bIsWorkflowPending
    End Sub

    Private Sub onWorkflowItemChanged(ByVal sender As System.Object, ByVal e As System.Workflow.Activities.ExternalDataEventArgs)
        CheckStatus()
    End Sub

    Private Sub CheckStatus()
        If CStr(workflowProperties.Item("Document Status")) = "Review Complete" Then
            bIsWorkflowPending = False
        End If
    End Sub
End Class




